### PR TITLE
Deepgram - New Components

### LINKED ISSUE #6271: [ACTION] Deepgram Actions
**Is there a specific app this action is for?**
Deepgram

**Please provide a link to the relevant API docs for the specific service / operation.**
[There are a few different actions that may be useful for our users.](https://developers.deepgram.com/api-reference/)

**Transcription** https://developers.deepgram.com/api-reference/transcription/#transcribe-pre-recorded-

1. Transcribe pre-recorded audio audio

Please include all options - tier, model, diarize, punctuate, numerals, etc

**Projects** https://developers.deepgram.com/api-reference/projects/#get-projects

1. Get Projects 
2. Get Project 

**Usage** https://developers.deepgram.com/api-reference/usage/#summarize-usage 

1. Get all requests 
2. Get Request
3. Summarize Usage 

**Billing** https://developers.deepgram.com/api-reference/billing/#get-balance

1. Get all balances
2. Get Balance

### PR DESCRIPTION
**Pull request by michelle0927**
Resolves #6271

**Actions**

- Transcribe Audio
- List Projects
- Get Project
- List Requests
- Get Request
- Summarize Usage
- List Balances
- Get Balance

### CODE REVIEW

File: components/deepgram/actions/get-request/get-request.mjs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  },
+  async run({ $ }) {
+    const request = await this.deepgram.getRequest({
+      projectId: this.projectId,
+      requestId: this.requestId,
+      $,
+    });
+
+    if (request) {
+      $.export("$summary", `Successfully retrieved balance with ID ${this.requestId}`);
```
Comment (alysonturing) [inline components/deepgram/actions/get-request/get-request.mjs:35]:
```suggestion
      $.export("$summary", `Successfully retrieved request with ID ${this.requestId}`);
```

File: components/deepgram/actions/transcribe-audio/transcribe-audio.mjs [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+      label: "Detect Entities",
+      description: "Indicates whether Deepgram will identify and extract key entities for sections of content. When `detect_entities` is set to `true`, `punctuate` will be set to true by default.",
+      optional: true,
+    },
+    utterances: {
+      type: "boolean",
+      label: "Utterances",
+      description: "Indicates whether Deepgram will segment speech into meaningful semantic units, which allows the model to interact more naturally and effectively with speakers’ spontaneous speech patterns",
+      optional: true,
+
```
Comment (alysonturing) [inline components/deepgram/actions/transcribe-audio/transcribe-audio.mjs:154]:
```suggestion
```

File: components/deepgram/deepgram.app.mjs [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+        }) => ({
+          label: path,
+          value: id,
+        })) || [];
+      },
+    },
+    language: {
+      type: "string",
+      label: "Language",
+      description: "BCP-47 language tag that hints at the primary spoken language",
```
Comment (alysonturing) [inline components/deepgram/deepgram.app.mjs:57]: Can you provide a link with docs about `BCP-47` or an exemple of how to fill the prop like e.g. `en-US`

Comment (alysonturing) [reply components/deepgram/deepgram.app.mjs:57]: or options / async options

**Review by alysonturing — APPROVED** [review]
LGTM! there's a typo in one export, also, I made a suggestion about BCP-47 languages, please check this before merging.

**Comment by vunguyenhung** [comment]
Hi everyone, all test cases are passed! Ready for release! 

Test report
[Deepgram_6303_2351.pdf](https://github.com/PipedreamHQ/pipedream/files/11423250/Deepgram_6303_2351.pdf)

**Review by GTFalcao — APPROVED** [review]

**Pull request merged**